pkg/scale/decode.go

```diff
@@ -93,15 +90,17 @@ type Decoder struct {
 func (d *Decoder) Decode(dst interface{}) (err error) {
 	dstv := reflect.ValueOf(dst)
 	if dstv.Kind() != reflect.Ptr || dstv.IsNil() {
-		err = fmt.Errorf("unsupported dst: %T, must be a pointer to a destination", dst)
+		err = fmt.Errorf("%w: %T", ErrUnsupportedDestination, dst)
 		return
 	}
 
 	elem := indirect(dstv)
+
+	err = d.unmarshal(elem)
 	if err != nil {
-		return
+		return fmt.Errorf("unmarshalling %#v: %w", elem, err)
 	}
-	return d.unmarshal(elem)
+	return
```

Comment:
nit let's make it explicit

```suggestion
	return nil
```